1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy Полученные значения должны быть равны. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [28]:
import pandas as pd

In [40]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
n=10

In [41]:
cov_ = (zp*ks).sum()/(n-1) - (zp.sum()/n)*(ks.sum()/(n-1))
cov_

10175.377777777758

In [35]:
r = cov_ /(zp.std()*ks.std())
r

0.8874900920739158

In [36]:
np.cov(zp,ks)

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

Полученные значения совпадают (10175.37777778)

In [44]:
df = pd.DataFrame({"zp": zp, "ks": ks})
df

zp   ks
0   35  401
1   45  574
2  190  874
3  200  919
4   40  459
5   70  739
6   54  653
7  150  902
8  120  746
9  110  832

In [46]:
df.corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

Полученные значения совпадают (0.88749)

2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [1]:
import numpy as np

$\delta = \frac{t_{\lambda} \cdot \sigma}{\sqrt{n}}$

In [2]:
a = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [3]:
a_mean = a.mean()
a_mean

118.1

In [6]:
a_std = a.std(ddof=1)
a_std

10.54566788359614

In [8]:
n = len(a)
n

10

t_kr ищем по таблице Стьюдента, степень свободы = 9

In [10]:
t_kr = 2.262

In [15]:
delta = t_kr * a_std / (n**0.5)
delta

7.543392236918348

In [16]:
a_mean - delta

110.55660776308164

In [17]:
a_mean + delta

125.64339223691834

Ответ ДИ = (110.5566, 125.6433)

3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [18]:
d = 25

In [19]:
n = 27

In [20]:
a_mean = 174.2

$\delta = \frac{t_{\lambda} \cdot \sigma}{\sqrt{n}}$

t_lambda ищем по таблице Лапласа, lamdba = 0.95, 2Ф(lambda) = 0.95, Ф(lambda) = 0.475

In [23]:
t_lambda = 1.96

In [24]:
delta = t_lambda * (d ** 0.5) / (n ** 0.5)
delta

1.8860108793527777

In [25]:
a_mean - delta

172.31398912064722

In [26]:
a_mean + delta

176.08601087935276

Ответ ДИ = (172.31, 176.086)